pkg/errors/errors.go

```diff
@@ -21,12 +21,23 @@ import (
 	"unicode/utf8"
 )
 
-func NewInvalidCRD(apiGroup string) *KNError {
+func NewInvalidCRD(apiGroup string, canFindResource string) *KNError {
```

Comment:
How about creating a specific cause functions per message. E.g. `NewInvalidCRDCauseUnknown`, or define cause as a const string? 

It might be easier to understand and consume whenever such new error is going to be added elsewhere etc.